### ISSUE TITLE
Spamming "No incoming ion beam" warnings

### ISSUE THREAD

**Issue by DraTeots** [eic/EICrecon#262]
While running eicrecon with particle gun single particles I get a lot of spam with: 

```
[...][Digi:SmearedFarForwardParticles] [warning] No incoming ion beam; using 100 GeV ion beam energy
```

While the warning is pretty important in cases where it is not possible to identify incoming beam particles for generated events, there are many cases where it is not determined by the very nature of run (like with particle gun) and should not be spammed. 

My proposed fix would be to call the warning only once. 

<img width="1006" alt="image" src="https://user-images.githubusercontent.com/4732545/197271362-b1331779-8f25-4dd4-b095-50bc41da1fc2.png">

**Comment by DraTeots** [comment]
Workaround

```
-PDigi:SmearedFarForwardParticles:LogLevel=error
```

**Comment by faustus123** [comment]
This was fixed in PR#178 by limiting it to print the warning only once.
